app/features/build_runner/build_runner.rb

```diff
@@ -78,10 +89,11 @@ def initialize(project:, sha:, github_service:, notification_service:, work_queu
 
       prepare_build_object(trigger: trigger)
 
+      local_folder = @local_build_folder || File.join(project.local_repo_path, "builds", sha)
       @repo = GitRepo.new(
         git_config: project.repo_config,
         provider_credential: github_service.provider_credential,
-        local_folder: File.join(project.local_repo_path, "builds", sha),
```

Comment:
don't always assume we're using the `builds` folder, we might have a 1-off build